tests/TestCase/ORM/AssociationTest.php

```diff
@@ -142,6 +142,31 @@ public function testClassNameUnnormalized()
     }
 
     /**
+     * Tests that an exception is thrown when invalid target table is fetched
+     * from a registry.
+     *
+     * @return void
+     * @expectedException \RuntimeException
+     */
+    public function testInvalidTableFetchedFromRegistry()
+    {
+        TableRegistry::get('Test');
+
+        $config = [
+            'className' => '\Cake\Test\TestCase\ORM\TestTable',
```

Comment:
It exists but 'TableRegistry' has already registered a table with alias 'Test' which is an instance of 'Cake\ORM\Table'. In an association we're requesting a 'Cake\Test\TestCase\ORM\TestTable' instance with 'Test' alias but the registry will get us that already registered instance which is not what we want.